include/swift/Runtime/Once.h

```diff
@@ -27,6 +27,11 @@ namespace swift {
 // On OS X and iOS, swift_once_t matches dispatch_once_t.
 typedef long swift_once_t;
 
+#elif defined(__CYGWIN__)
+
+// On Cygwin, std::once_flag can not be used because it is larger than the
+// platform word.
+typedef unsigned long swift_once_t;
```

Comment:
It seems other implementations (Linux, Cygwin internal.. ) didn't use a pointer type, so I chosen that.